ddtrace/contrib/aioredis/patch.py

```diff
@@ -0,0 +1,172 @@
+
+
+def patch():
+    if getattr(aioredis, "_datadog_patch", False):
+        return
+    setattr(aioredis, "_datadog_patch", True)
+    pin = Pin()
+    if aioredis_version >= (2, 0):
+        _w("aioredis.client", "Redis.execute_command", traced_execute_command)
+        _w("aioredis.client", "Redis.pipeline", traced_pipeline)
+        _w("aioredis.client", "Pipeline.execute", traced_execute_pipeline)
+        pin.onto(aioredis.client.Redis)
+    else:
+        _w("aioredis", "Redis.execute", traced_13_execute_command)
+        _w("aioredis", "Redis.pipeline", traced_13_pipeline)
+        _w("aioredis.commands.transaction", "Pipeline.execute", traced_13_execute_pipeline)
+        pin.onto(aioredis.Redis)
+
+
+def unpatch():
+    if not getattr(aioredis, "_datadog_patch", False):
+        return
+
+    setattr(aioredis, "_datadog_patch", False)
+    if aioredis_version >= (2, 0):
+        _u(aioredis.client.Redis, "execute_command")
+        _u(aioredis.client.Redis, "pipeline")
+        _u(aioredis.client.Pipeline, "execute")
+    else:
+        _u(aioredis.Redis, "execute")
+        _u(aioredis.Redis, "pipeline")
+        _u(aioredis.commands.transaction.Pipeline, "execute")
+
+
+async def traced_execute_command(func, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return await func(*args, **kwargs)
+
+    decoded_args = [arg.decode() if isinstance(arg, bytes) else arg for arg in args]
+    with _trace_redis_cmd(pin, config.aioredis, instance, decoded_args):
+        return await func(*args, **kwargs)
+
+
+async def traced_pipeline(func, instance, args, kwargs):
+    pipeline = await func(*args, **kwargs)
+    pin = Pin.get_from(instance)
+    if pin:
+        pin.onto(pipeline)
+    return pipeline
+
+
+async def traced_execute_pipeline(func, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return await func(*args, **kwargs)
+
+    cmds = [format_command_args(c) for c, _ in instance.command_stack]
+    resource = "\n".join(cmds)
+    with _trace_redis_execute_pipeline(pin, config.aioredis, resource, instance):
+        return await func(*args, **kwargs)
+
+
+def traced_13_pipeline(func, instance, args, kwargs):
+    pipeline = func(*args, **kwargs)
+    pin = Pin.get_from(instance)
+    if pin:
+        pin.onto(pipeline)
+    return pipeline
+
+
+def traced_13_execute_command(func, instance, args, kwargs):
+    # If we have a _RedisBuffer then we are in a pipeline
+    if isinstance(instance.connection, _RedisBuffer):
+        return func(*args, **kwargs)
+
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return func(*args, **kwargs)
+
+    decoded_args = [arg.decode() if isinstance(arg, bytes) else arg for arg in args]
+    span = pin.tracer.trace(
+        redisx.CMD, service=trace_utils.ext_service(pin, config.aioredis), span_type=SpanTypes.REDIS
```

Comment:
```suggestion
        redisx.CMD, service=trace_utils.ext_service(pin, config.aioredis), span_type=SpanTypes.REDIS, activate=False
```

Context management here is tricky. If another `execute` call is made after this one then it will become a child of this operation which isn't really true. It would probably be better represented as a sibling operation under the same parent hence we don't want to activate the span.

This works assuming there are no operations/spans beneath this one (which I don't think there should be?)


We should also add a test case to verify this

Reply:
Is this test case what you had in mind: f72a6f2045f53fef27577640b65fb974ce685e55